Sources/TokamakCore/Fiber/Fiber.swift

```diff
@@ -0,0 +1,240 @@
+//
+//  File.swift
+//
+//
+//  Created by Carson Katri on 2/15/22.
+//
+
+@_spi(TokamakCore) public extension FiberReconciler {
+  /// A manager for a single `View`.
+  ///
+  /// There are always 2 `Fiber`s for every `View` in the tree,
+  /// a current `Fiber`, and a work in progress `Fiber`.
+  /// They point to each other using the `alternate` property.
+  ///
+  /// The current `Fiber` represents the `View` as it is currently rendered on the screen.
+  /// The work in progress `Fiber` (the `alternate` of current), is used in the reconciler to compute the new tree.
+  ///
+  /// When reconciling, the tree is recomputed from the root of the state change on the work in progress `Fiber`.
+  /// Each node in the fiber tree is updated to apply any changes, and a list of mutations needed to get the
+  /// rendered output to match is created.
+  ///
+  /// After the entire tree has been traversed, the current and work in progress trees are swapped,
+  /// making the updated tree the current one, and leaving the previous current tree available to apply future changes on.
+  final class Fiber: CustomDebugStringConvertible {
+    weak var reconciler: FiberReconciler<Renderer>?
+
+    /// The underlying `View` instance.
+    var view: Any!
+    /// Outputs from evaluating `View._makeView`
+    var outputs: ViewOutputs!
+    /// A function to visit `view` generically.
+    var visitView: ((ViewVisitor) -> ())!
+    /// The identity of this `View`
+    var id: Identity?
+    /// The mounted element, if this is a Renderer primitive.
+    var element: Renderer.ElementType?
+    /// The first child node.
+    var child: Fiber?
+    /// This node's right sibling.
+    var sibling: Fiber?
+    /// An unowned reference to the parent node.
+    unowned var parent: Fiber?
+    /// The nearest parent that can be mounted on.
+    unowned var elementParent: Fiber?
```

Comment:
Any specific reason this is `unowned` instead of `weak`? I think using the former could be fine, as long as the nuance is explained here in code comments.

Reply:
I added a doc comment explaining `unowned` on the `parent` property. If you think my reasoning behind it is flawed though, let me know and I'll switch it to `weak`. I'm always a bit unsure of myself on ARC stuff 😅